xtc/src/history.rs

```diff
@@ -50,7 +49,7 @@ impl HistoryBuffer {
                 return 0;
             }
 
-            trap("Transaction is expected to have a non-zero amount.")
```

Comment:
not sure about this change: api::trap is defined in the cdk and currently calls into ic0::trap which seems to be a C function imported into the webassembly module.

do we want to assume that panic! is doing the same thing as trap calling into ic0::trap? Also what if at some point the cdk changes api::trap's implementation to do something before calling into ic0::trap

sorry if I am being too cautious here

Reply:
panic also traps the canister, they have the same effect.